docs: add documentation for all secure acl introduction work (#5640)
hashicorp/consul: website/source/api/acl/tokens.html.md

  - `ServiceName` `(string: <required>)` - The name of the service. The name
    must be no longer than 256 characters, must start and end with a lowercase
    alphanumeric character, and can only contain lowercase alphanumeric
    characters as well as `-` and `_`.

  - `Datacenters` `(array<string>)` - Specifies the datacenters the effective
    policy is valid within. When no datacenters are provided the effective
    policy is valid in all datacenters including those which do not yet exist
    but may in the future.

- `Local` `(bool: false)` - If true, indicates that the token should not be
  replicated globally and instead be local to the current datacenter.

- `ExpirationTime` `(time: "")`- If set this represents the point after which a
  token should be considered revoked and is eligible for destruction. The
  default unset value represents NO expiration. This value must be between 1
  minute and 24 hours in the future. Added in Consul 1.5.0.

- `ExpirationTTL` `(duration: 0s)` - This is a convenience field and if set
  will initialize the `ExpirationTime` field to a value of `CreateTime +
  ExpirationTTL`. This field is not persisted beyond its initial use. Can be

>   specified in the form of `"10s"` or `"5m"` (i.e., 10 seconds or 5 minutes,

Reviewer: ```suggestion
  specified in the form of `"60s"` or `"5m"` (i.e., 60 seconds or 5 minutes,
```

Reviewer: If the time can't be smaller than a minute the example should be a valid value.

   internally be resolved to the policy ID. With linking tokens internally by IDs,
   Consul enables policy renaming without breaking tokens.

- `Roles` `(array<RoleLink>)` - The list of roles that should be applied to the
   token. A RoleLink is an object with an "ID" and/or "Name" field to specify a
   role. With the RoleLink, tokens can be linked to roles either by the role
   name or by the role ID. When roles are linked by name they will be internally
   resolved to the role ID. With linking tokens internally by IDs, Consul
   enables role renaming without breaking tokens.

~> TODO:where is the best place to mention RoleLink.BoundName?

> - `ServiceIdentities` `(array<ServiceIdentity>)` - The list of service

Reviewer 2: Again, maybe we could link out here to the `service identity definition`

- `Description` `(string: "")` - Free form human readable description of this token.

- `Policies` `(array<PolicyLink>)` - This is the list of policies that should
  be applied to this token. A PolicyLink is an object with an "ID" and/or

>   "Name" field to specify a policy. With this tokens can be linked to policies

Reviewer: ```suggestion
  "Name" field to specify a policy. With the PolicyLink tokens can be linked to policies
```

  either by the policy name or by the policy ID. When policies are linked by
  name they will internally be resolved to the policy ID. With linking tokens
  internally by IDs, Consul enables policy renaming without breaking tokens.

- `Roles` `(array<RoleLink>)` - The list of roles that should be applied to the
  token. A RoleLink is an object with an "ID" and/or "Name" field to specify a
  role. With the RoleLink, tokens can be linked to roles either by the role
  name or by the role ID. When roles are linked by name they will be internally
  resolved to the role ID. With linking tokens internally by IDs, Consul
  enables role renaming without breaking tokens.

- `ServiceIdentities` `(array<ServiceIdentity>)` - The list of [service
  identities](/docs/acl/acl-system.html#acl-service-identities) that should be

>   applied to the token.  Added in Consul 1.5.0.

Reviewer: ```suggestion
  applied to the token. Added in Consul 1.5.0.
```

### Sample Request

> ```text

Reviewer 2: ```suggestion
```sh
```
